core/shared/src/main/scala/zio/prelude/These.scala

```diff
@@ -31,6 +31,14 @@ sealed trait These[+A, +B] { self =>
       case (Both(l, r1), Right(r2))     => Both(l, B1.combine(r1, r2))
       case (Both(l1, r1), Both(l2, r2)) => Both(A1.combine(l1, l2), B1.combine(r1, r2))
     }
+
+  def fold[A1 >: A](f: A => A1, g: B => A1)(implicit A1: Associative[A1]): A1 =
```

Comment:
```suggestion
  def fold[A1](f: A => A1, g: B => A1)(implicit A1: Associative[A1]): A1 =
```
Unless I'm missing something, the result type can be anything.

Reply:
That is 💯  right. There is another variant `foldAssociative` that would not take any user provided function and require `A1 >: A` but for this version I would probably call it `C` for clarity and it can be completely unconstrained.

Reply:
@regiskuckaertz thanks for catching that.
Not sure what I was thinking 🤔

Reply:
updated